src/inserter.rs

```diff
@@ -199,6 +202,16 @@ where
         self.ticks.reschedule();
     }
 
+    /// Registers a callback that will be invoked after each successful batch commit.
+    ///
+    /// The callback receives the committed [`Quantities`]. It is invoked only
+    /// when a batch actually commits (i.e., non-zero transactions), and only
+    /// after the commit completes successfully.
+    pub fn with_on_commit(mut self, callback: impl FnMut(&Quantities) + Send + 'static) -> Self {
```

Comment:
Hmm, `with_on_commit` is a little clunky but the `with_` prefix does match the other combinators.

Maybe `with_commit_callback` or just `on_commit`? I don't feel particularly strongly about any of these, though.